pkg/metrics/collector/pod_ip_metrics.go

```diff
@@ -139,7 +158,29 @@ func init() {
 // NewPodIpMetricsCollector returns a new Collector exposing pod IP allocation
 // stats.
 func NewPodIPMetricsCollector() (Collector, error) {
-	return &podIPMetricsCollector{clock: &realClock{}}, nil
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		return nil, fmt.Errorf("error creating in-cluster config: %v", err)
+	}
+
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return nil, fmt.Errorf("error creating clientset: %v", err)
+	}
+
+	nodeName, err := os.Hostname()
+	if err != nil {
+		return nil, fmt.Errorf("error getting hostname: %v", err)
+	}
+	if nodeName == "" {
+		return nil, fmt.Errorf("HOSTNAME environment variable not set")
```

Comment:
https://cs.opensource.google/go/go/+/refs/tags/go1.20.3:src/os/sys_linux.go it's not based on env. Will it return empty anyway (I'm not sure)?